src/components/ButtonLink.tsx

```diff
@@ -1,24 +1,13 @@
 import React from "react"
-import { Button, ButtonProps, useStyleConfig } from "@chakra-ui/react"
 
+import type { IProps as IButtonProps } from "./Button"
 import Link, { IBaseProps as ILinkProps } from "./Link"
+import Button from "./Button"
 
-export interface IProps extends ILinkProps, ButtonProps {}
+export interface IProps extends ILinkProps, Omit<IButtonProps, "toId"> {}
```

Comment:
Improve type safety by emitting a TypeError for `toId`, since this component renders a link.